Sources/SwiftWarningControl/WarningControlRegions.swift

```diff
@@ -0,0 +1,329 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// A single warning control region, consisting of a start and end positions,
+/// a diagnostic group identifier, and an emission behavior specifier.
+@_spi(ExperimentalLanguageFeatures)
+public struct WarningControlRegion {
```

Comment:
This type will be a little useless as public API if none of its members are public 😉